pkg/flare/flare.go

```diff
@@ -32,41 +31,48 @@ type flareResponse struct {
 
 // SendFlareWithHostname sends a flare with a set hostname
 func SendFlareWithHostname(archivePath string, caseID string, email string, hostname string) (string, error) {
-	body := &bytes.Buffer{}
-	writer := multipart.NewWriter(body)
-
-	p, err := writer.CreateFormFile("flare_file", filepath.Base(archivePath))
-	if err != nil {
-		return "", err
-	}
-	file, err := os.Open(archivePath)
-	defer file.Close()
-	if err != nil {
-		return "", err
-	}
-	_, err = io.Copy(p, file)
-	if err != nil {
-		return "", err
-	}
-	if caseID != "" {
-		writer.WriteField("case_id", caseID)
-	}
-	if email != "" {
-		writer.WriteField("email", email)
-	}
-
-	// Send the full version
-	av, _ := version.New(version.AgentVersion, version.Commit)
-	writer.WriteField("agent_version", av.String())
-	writer.WriteField("hostname", hostname)
-
-	err = writer.Close()
-	if err != nil {
-		return "", err
-	}
+	bodyReader, bodyWriter := io.Pipe()
+	writer := multipart.NewWriter(bodyWriter)
+
+	//Write stuff to the pipe will block until it is read from the other end, so we don't load everything in memory
+	go func() {
+
+		defer bodyWriter.Close()
+		defer writer.Close()
```

Comment:
tiny nite: ideally we'd want to close `writer` before we close `bodyWriter`

Reply:
That's what we do: `defer` calls get stacked, so the last `defer` you call is the first one that gets executed.

Reply:
True!